src/Mvc/test/Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation.Test/CSharpCompilerTest.cs

```diff
@@ -0,0 +1,415 @@
+            var compiler = new CSharpCompiler(ReferenceManager, hostingEnvironment);
+
+            // Act & Assert
+            var compilationOptions = compiler.CSharpCompilationOptions;
+            Assert.False(compilationOptions.AllowUnsafe);
+            Assert.Equal(ReportDiagnostic.Default, compilationOptions.GeneralDiagnosticOption);
+            Assert.Equal(OptimizationLevel.Debug, compilationOptions.OptimizationLevel);
+            Assert.Collection(compilationOptions.SpecificDiagnosticOptions.OrderBy(d => d.Key),
+                item =>
+                {
+                    Assert.Equal("CS1701", item.Key);
+                    Assert.Equal(ReportDiagnostic.Suppress, item.Value);
+                },
+                item =>
+                {
+                    Assert.Equal("CS1702", item.Key);
+                    Assert.Equal(ReportDiagnostic.Suppress, item.Value);
+                },
+                item =>
+                {
+                    Assert.Equal("CS1705", item.Key);
+                    Assert.Equal(ReportDiagnostic.Suppress, item.Value);
+                });
+        }
+
+        [Fact]
+        public void EnsureOptions_ConfiguresDefaultParseOptions()
+        {
+            // Arrange
+            var hostingEnvironment = Mock.Of<IHostingEnvironment>(h => h.EnvironmentName == "Development");
+            var compiler = new CSharpCompiler(ReferenceManager, hostingEnvironment);
+
+            // Act & Assert
+            var parseOptions = compiler.ParseOptions;
+            Assert.Equal(LanguageVersion.CSharp7, parseOptions.LanguageVersion);
+            Assert.Equal(new[] { "DEBUG" }, parseOptions.PreprocessorSymbolNames);
+        }
+
+        [Fact]
+        public void Constructor_ConfiguresPreprocessorSymbolNames()
+        {
+            // Arrange
+            var hostingEnvironment = Mock.Of<IHostingEnvironment>();
+            var dependencyContextOptions = new DependencyContextCompilationOptions(
+               new[] { "SOME_TEST_DEFINE" },
+               languageVersion: null,
+               platform: null,
+               allowUnsafe: null,
+               warningsAsErrors: true,
+               optimize: null,
+               keyFile: null,
+               delaySign: null,
+               publicSign: null,
+               debugType: null,
+               emitEntryPoint: null,
+               generateXmlDocumentation: null);
+
+            var compiler = new TestCSharpCompiler(ReferenceManager, hostingEnvironment, dependencyContextOptions);
+
+            // Act & Assert
+            var parseOptions = compiler.ParseOptions;
+            Assert.Contains("SOME_TEST_DEFINE", parseOptions.PreprocessorSymbolNames);
+        }
+
+        [Fact]
+        public void Constructor_ConfiguresLanguageVersion()
+        {
+            // Arrange
+            var dependencyContextOptions = new DependencyContextCompilationOptions(
```

Comment:
This is repeated many times. Suggest using a helper to make it clear where the options are different.